Conda's dictionary packages install to share/hunspell_dictionaries.
Resolve that directory relative to the library so relocated environments work.

--- a/providers/enchant_hunspell.cpp
+++ b/providers/enchant_hunspell.cpp
@@ -205,6 +205,12 @@ s_buildDictionaryDirs (EnchantProvider * me, std::vector<std::string> & dirs)
 	dirs.push_back (tmp);
 	g_free(tmp);
 
+	gchar * prefix = enchant_get_prefix_dir ();
+	tmp = g_build_filename (prefix, "share", "hunspell_dictionaries", nullptr);
+	g_free (prefix);
+	dirs.push_back (tmp);
+	g_free (tmp);
+
 	for (const gchar* const * iter = g_get_system_data_dirs (); *iter; iter++)
 		{
 			tmp = g_build_filename (*iter, me->identify (me), nullptr);
